server/src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -101,20 +102,46 @@ public static void execute(SearchContext context) {
 
         Aggregator[] aggregators = context.aggregations().aggregators();
 
-        List<InternalAggregation> aggregations = new ArrayList<>(aggregators.length);
         if (context.aggregations().factories().context() != null) {
             // Rollup can end up here with a null context but not null factories.....
             context.aggregations().factories().context().multiBucketConsumer().reset();
         }
+        // decide if we're in new style or old style
+        boolean useNewStyle = true;
         for (Aggregator aggregator : context.aggregations().aggregators()) {
+            useNewStyle = useNewStyle && aggregator.canUseCollectedAggregator();
+        }
+        if (useNewStyle) {
```

Comment:
In a perfect world, this never merges to main.  This is the minimum logic to kind-of support mixed mode operation on a single cluster, but to do that in production we'd want a lot more testing, at a minimum.  I'd rather live in a world where all aggregations are either old style or new style, and we cleanly cut over at a version boundary.